### PR TITLE
[JUJU-1633] Rewrite api/client/cloud unit tests to use gomock

### PR DESCRIPTION
**Pull request by cderici**
The unit tests for api/client/cloud now use go mock and not juju/testing. All unit tests should pass.

### CODE REVIEW

File: api/client/cloud/cloud_test.go [hunk 1]
```diff
@@ -209,196 +180,197 @@ func (s *cloudSuite) TestClouds(c *gc.C) {
...
+	res := new(params.UpdateCredentialResults)
+	results := params.UpdateCredentialResults{
+		Results: []params.UpdateCredentialResult{
+			{
+				CredentialTag: "cloudcred-foo_bob_bar",
+				Error:         apiservererrors.ServerError(errors.New("validation failure")),
 			},
-		),
-		BestVersion: 3,
+		},
```
Comment (hmlanigan) [inline api/client/cloud/cloud_test.go:297]: removing the `BestVersion: 3`, the scope of the test has changed. It appears that version 3 of the facade was removed in juju 3.0. Is `TestUpdateCredentialCallError` really different from `TestUpdateCredentialError`?

Comment (hmlanigan) [reply api/client/cloud/cloud_test.go:297]: Same for any test where `BestVersion: 3` has been removed.

**Comment by cderici** [comment]
> @cderici do you plan to squash the 3 commits into one before merging? If not, the commit messages provide no understanding what is different between them.

I wasn't planning on doing that but yeah I agree that might be a good idea, thanks for the heads up :+1:

**Review by hmlanigan — APPROVED** [review]
LGTM

**Pull request merged**